[davidgohzk] iP

davidgohzk opened the pull request:
# Barney
Barney is a ~dinosaur~ is your own personal dinosaur chat bot that you can use to keep track of the things that you need to do.

> Barney allowed me to USE MY IMGAINATION to good use and now I am 10x more productive - a happy customer

- text-based
- easy to learn or use
- _FULL OF IMAGINATION_

## What you need to do to get it to run
1. Download the latest JAR here: https://github.com/davidgohzk/ip/releases/tag/A-Jar
2. Ensure that you have Java 17 JRE installed on your system
3. Double click and let it **ROAR**

If you are a Java Programmer, you can add new tasks at the `src/main/java/barney/action/CommandManager.java`


```java
LIST("list", "^list\\s*$", ListCommand.class), MARK("mark", "^mark\\s+\\d+\\s*$", MarkCommand.class, "index"),
UNMARK("unmark", "^unmark\\s+\\d+\\s*", UnmarkCommand.class, "index"),
TODO("todo", "^todo\\s+.+\\s*", TodoCommand.class, "description"),
DEADLINE("deadline", "^deadline\\s+.+\\s+/by\\s+.+\\s*$", DeadlineCommand.class, "description", "by"),
EVENT("event", "^event\\s+.+\\s+/from\\s+.+\\s+/to\\s+.+\\s*$", EventCommand.class, "description", "from",
    "to"),
DELETE("delete", "^delete\\s+\\d+\\s*$", DeleteCommand.class, "index"),
FIND("find", "^find\\s+.+\\s*$", FindCommand.class, "keyword"),
BYE("bye", "^bye\\s*$", EndCommand.class);
```
The format is 
```
NAME_OF_COMMAND(
    "<String that is used for the command>",
    Regex that is used to detect,
    Class that is used to handle the command,
    the arguments for the command
)
```

## Features
- [X] Managing Tasks
- [x] Managing Deadlines
- [ ] GUI (Coming Soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

davidgohzk committed:
added ignore .class files in .gitignore

davidgohzk committed:
renamed from Duke to Barney

davidgohzk committed:
put the text into functions

davidgohzk committed:
added echo functionality

davidgohzk committed:
added add and list functionality

davidgohzk committed:
changes from an arraylist into a array of strings for the list

davidgohzk committed:
added mark as done functionality

davidgohzk committed:
added different tasks

davidgohzk committed:
updated test to include the tests

davidgohzk committed:
added in error handling

davidgohzk committed:
added in delete functionality

davidgohzk committed:
added saving code

davidgohzk committed:
added datetime to the app

davidgohzk committed:
OOP

davidgohzk committed:
OOP

davidgohzk committed:
following style guide

davidgohzk committed:
refactored the files to make it more higher order

davidgohzk committed:
moved the files into packages

davidgohzk committed:
Merge remote-tracking branch 'origin/add-gradle-support'

davidgohzk committed:
changed the project path to Barney

davidgohzk committed:
refactored into a single folder

davidgohzk committed:
refactored into a single folder

davidgohzk committed:
add JUnit

davidgohzk committed:
add filename to build.grade for jar

davidgohzk committed:
add javadocs to files

davidgohzk committed:
add javadocs

davidgohzk committed:
add javadocs

davidgohzk committed:
add find functionality

davidgohzk committed:
resolved conflict

davidgohzk committed:
Reseting for git commit

davidgohzk committed:
added styling

davidgohzk committed:
add extra tests

davidgohzk committed:
change some code to make it more readable and better code quality

davidgohzk committed:
add checkstyle code

davidgohzk committed:
add changes for coding standard

davidgohzk committed:
handle git merge

davidgohzk committed:
resolve issue with merge

davidgohzk committed:
resolve issue with merge

davidgohzk committed:
Merge branch 'branch-A-Checkstyle'

davidgohzk committed:
resolve checkstyle issues

chenle228 commented on src/main/java/Barney.java:
```diff
@@ -1,8 +1,21 @@
+        String text = "Goodbye, I am Barney <RAWR>, see you next time!";
+        System.out.println(text);
+        System.out.println(LONG_LINE);
+    }
```
I really like how you refactored the code into reusable functions right from the start! It really demonstrated how you kept OOP design principle in mind, especially at this level the instructions hadn't asked for refactoring yet:)

davidgohzk replied:
Thank you so much!

chenle228 commented on src/main/java/Barney.java:
```diff
@@ -12,6 +16,24 @@ public static void welcomeText() {
+        }
+    }
+
     public static void endingText() {
```
Just a very minor issue here, perhaps the `endingText` method here could be renamed to `endText` to align with the coding standard?

chenle228 commented on src/main/java/Barney.java:
```diff
@@ -17,18 +18,28 @@ public static void welcomeText() {
     public static void chat() {
-        String input = "";
+        String input;
         Boolean isChatting = true;
```
Good naming of boolean variable that follows the coding standard!

davidgohzk replied:
Thank you!

chenle228 commented on src/main/java/Barney.java:
```diff
@@ -1,55 +1,89 @@
+            String keyword = inputArray[0];
+
+            switch (keyword) {
                 case "list":
```
Based on the coding standard, we don't need indentation for `case` clauses, so perhaps the indentation for all `case` clauses in the switch statement could be removed?

davidgohzk replied:
Gotcha

chenle228 commented on src/main/java/Barney.java:
```diff
@@ -1,55 +1,89 @@
-                        System.out.println(i + ". " + list[i - 1]);
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < listLength; i++) {
+                        System.out.println(String.format("%d. %s", i + 1, taskList[i].toString()));
```
Perhaps these 3 lines of code could be indented since they are within the first loop?

chenle228 reviewed (commented):
Overall I found your code well-thought-out and well-written! Keep up the good work:)

davidgohzk committed:
fix bug of off by 1 error in argument map size

davidgohzk committed:
fix bug of off by 1 error in argument map size

davidgohzk committed:
add javafx dependencies into the project

davidgohzk committed:
refactor to add in some basic gui

davidgohzk committed:
add tutorial files to this project

davidgohzk committed:
add in the barney specific implementations

davidgohzk committed:
fix the version differences

davidgohzk committed:
add comment

davidgohzk committed:
add assertions

davidgohzk committed:
Remove the redundant regex in CommandManager enum

Previously, there was a field the CommandType enum called regex that
required you to specify the regex for the command.

However, this can be done dynamically based on the arguments of the
commands that are present.

Hence, removed the regex field and dynamically create the regex based
on the argument list.

This helps to reduce code burden in needing to specify the regex, which
might get into errors

davidgohzk committed:
Refactor the argument parse in Command Manager.

Argument parsing was done using switch case on the length of the
argument list. However, this can be done dynamically through code.

Changed from switch case to a for loop.

Makes the reading more clear and reduces the dependency on needing the
manage the switch case when coding.

davidgohzk committed:
Merge branch 'branch-A-Assertions'

davidgohzk committed:
Merge branch 'master' into branch-A-CodeQuality

davidgohzk committed:
Add streams to TaskLisk

davidgohzk committed:
Merge pull request #1 from davidgohzk/branch-A-Streams

Add streams to TaskLisk

davidgohzk committed:
Add branch C tagging command

davidgohzk committed:
Remove line length error

davidgohzk committed:
Add the AI.md

davidgohzk committed:
update userguide
